src/utilities.py

```diff
@@ -86,8 +86,14 @@ def set_chooser(chooser, choice):
 
 def populate_chooser(chooser, chooser_list, translate=True):
     """Fill the dropdown menu with the strings in a chooser_list"""
+    if chooser.get_model():
+        model = chooser.get_model()
+        for _item in model:
+            model.remove(0)
+    else:
```

Comment:
It somehow crashes when setting a new model to a chooser that already has an existing model. This was the main cause of the bug that this PR has been trying to solve. I haven't found a way to do this in a more pretty way, but this is quite simple nevertheless: If the model already exist, just empty it instead of creating a new one.